### ISSUE TITLE
`apim product create` parameter `--subscription-required` is not applied

### ISSUE THREAD

**Issue by oskrabanek** [Azure/azure-cli#23106]
## Describe the bug

**Command Name**
```
az apim product create `
            --resource-group $apimResourceGroup `
            --service-name $apimServiceName `
            --product-id "Product ID" `
            --product-name "Display name" `
            --description "Description" `
            --subscription-required $false `
            --state $product.state `
            --only-show-errors
```

**Errors:**
`subscription-required` is not applied in `create` nor `update` commands

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az apim product create --resource-group {} --service-name {} --product-id {} --product-name {} --description {} --subscription-required $false --state {} --only-show-errors`

## Expected Behavior
The `subscriptionRequired` property is set correctly

## Environment Summary
```
Windows-10-10.0.22000-SP0
Python 3.10.4
Installer: MSI

azure-cli 2.37.0
```
## Additional information
The `--approval-required` parameter might have the same issue

**Comment by yonzhan** [comment]
route to CXP team

**Comment by PramodValavala-MSFT** [comment]
@oskrabanek Looks like this has been fixed in the latest version 2.38.0. Try upgrading to the latest using `az upgrade` and try again.
